File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    def dfs(index, current_sum):
        # base case
        if index == len(numbers):
            return 1 if current_sum == target else 0
        
        # recursive case: + and - case
        return dfs(index+1, current_sum+numbers[index]) + dfs(index+1, current_sum - numbers[index])
        
    return dfs(0,0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;
vector<bool> visited; 


int solution(vector<int> numbers, int target) {
    int answer = 0;
    queue<int> q;
    q.push(0);
    
    for(int num: numbers){
        int size = q.size();
        
        for(int i=0; i<size; i++){           
            q.push(q.front()+num);
            q.push(q.front()-num);
            q.pop();
        }
    }
    while(!q.empty()){
        int x = q.front();
        if(x==target){
            answer++;
        }
        q.pop();
    }
    return answer;
}